docs/api-tutorial.md

```diff
@@ -0,0 +1,316 @@
+       --header 'Accept: application/json' \
+       --header 'Content-Type: application/json' \
+       --data '{"name": "Joe Bloggs"}' \
+       http://walkthrough.popit.mysociety.org/api/v0.1/persons
+
+    {
+      "result": {
+        "html_url": "http://walkthrough.popit.mysociety.org/persons/5373a60c657d191a1c3ff44e",
+        "url": "http://walkthrough.popit.mysociety.org/api/v0.1/persons/5373a60c657d191a1c3ff44e",
+        "id": "5373a60c657d191a1c3ff44e",
+        "name": "Joe Bloggs",
+        "memberships": [],
+        "links": [],
+        "contact_details": [],
+        "identifiers": [],
+        "other_names": []
+      }
+    }
+
+{% endhighlight %}
+
+There's a few things going on here so we'll walk through the options we passed to `curl` so you can see what they're all for:
+
+ * `--header 'Apikey: 2a0abbefe6aefb67ba5302a12570c2af2a7f4433'`
+
+ This is our apikey. You need this so that PopIt knows who you are and that you have permission to add data to your instance.
+
+ * `--header 'Accept: application/json'`
+
+ This tells PopIt that we'd like information back formatted as a JSON object
+
+ * `--header 'Content-Type: application/json'`
+
+ This tells PopIt that we're sending it data in the JSON format so it knows how to process it.
+
+ * `--data '{"name": "Joe Bloggs"}'`
+
+ This is the data we're uploading. This is the minimum information you need to add a person in PopIt
+
+ * `http://walkthrough.popit.mysociety.org/api/v0.1/persons`
+
+ And finally the API URL for persons.
+
+In the information returned by the API you'll notice that it's provided a URL for both the web page of your popit instance for the person - the `html_url` field - and also the API URL for the person - the `url` field. We'll use the second one later for updating the details.
+
+You can also see that PopIt generates an ID for the new person. If you want to use your own IDs then you can supply them and PopIt will use those instead:
+
+{% highlight bash %}
+
+$ curl --header 'Apikey: 2a0abbefe6aefb67ba5302a12570c2af2a7f4433' \
+       --header 'Accept: application/json' \
+       --header 'Content-Type: application/json' \
+       --data '{"id": "joe-bloggs", "name": "Joe Bloggs"}' \
+       http://walkthrough.popit.mysociety.org/api/v0.1/persons
+
+    {
+      "result": {
+        "html_url": "http://walkthrough.popit.mysociety.org/persons/joe-bloggs",
+        "url": "http://walkthrough.popit.mysociety.org/api/v0.1/persons/joe-bloggs",
+        "id": "joe-bloggs",
+        "name": "Joe Bloggs",
+        "memberships": [],
+        "links": [],
+        "contact_details": [],
+        "identifiers": [],
+        "other_names": []
+      }
+    }
+
+{% endhighlight %}
+
+Note that the `--data` line is now
+
+       --data '{"id": "joe-bloggs", "name": "Joe Bloggs"}'
+
+and that the `id` field and the two URLs are different. If we'd tried to use an existing ID PopIt would have returned an error.
+
+{% highlight json %}
```

Comment:
I think this needs to be `highlight bash`, otherwise the highlighting goes a bit wacky!